### DISCUSSION TITLE
[V5] AdonisV5, Pino Logs, and Pm2

### DISCUSSION THREAD

**Discussion by forzapa** [adonisjs/core#2514] [Help]
Hi, thanks for Adonis, it's really a great framework. 

I'm about to deploy in production my first application and I was wondering several things that have not yet been documented ( Can't wait for the documentation to be released ! 😄 )

I need to log a lot of information so I'm using pino with a custom transport. It works well, but with pm2 I'm having a hard time finding a solution to forward my logs.

So I have two questions: 
- Do you recommend using pm2 to deploy an Adonis application? If not, what do you recommend and how would you forward your logs with this solution?
- Do you have a solution to use pm2 and still forward the logs from Pino to Datadog or other?
Thanks a lot

**Comment by edesideri** [comment]
Hi,
im trying pm2 in these days and all works well, but if there are best solution for deploy im interesting too.
For logs i write in a mysql table so i can read the logs with a client application

**Reply by aayush123** [reply]
Even I have recently deployed an MVP to production using pm2. All seems fine, however, I do have to SSH into the server to see logs, if required.
I was thinking of using an external logging service soon, so even I'm looking forward to the discussion here.
Is PM2 ideal for adonis deployment? And how to further configure external logging?

**Comment by thetutlage** [comment]
Yes, I don't see anything wrong in using PM2. They have this guide on [managing logs](https://pm2.keymetrics.io/docs/usage/log-management/). Most probably you can write them in a file and configure datadog to read it from the file

**Comment by aayush123** [comment]
> Even I have recently deployed an MVP to production using pm2. All seems fine, however, I do have to SSH into the server to see logs, if required.
> I was thinking of using an external logging service soon, so even I'm looking forward to the discussion here.
> Is PM2 ideal for adonis deployment? And how to further configure external logging?

UPDATE:
I configured Papertrail on my production server to read pm2 logs.
It works pretty well :)
